### PR TITLE
Balance added

### PR DESCRIPTION
**Pull request by daniel-liemng**

### CODE REVIEW

**Comment by daniel-liemng** [comment]
Hi Aecio,
this branch just has 1 commit because earlier I deleted the .git file and re-initialize a new git. So this part include the sign up/login and balance page

File: client/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+node_modules/
```
Comment (aeciorc) [inline client/.gitignore:1]: I think this is already ignored from the root .gitignore

File: client/src/hocs/UnPrivateRoute.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import React, { useContext } from 'react';
+import { Route, Redirect } from 'react-router-dom';
+
+import { AuthContext } from '../context/AuthContext';
+
+const UnPrivateRoute = ({ component: Component, ...rest }) => {
```
Comment (aeciorc) [inline client/src/hocs/UnPrivateRoute.js:6]: Rename this to "PublicRoute" :P

File: client/src/context/UserContext.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  if (!user) return <Loading />;
+
+  return (
+    <UserContext.Provider value={{ user, setUser }}>
+      {children}
+    </UserContext.Provider>
+  );
+};
+
+export default UserProvider;
```
Comment (aeciorc) [inline client/src/context/UserContext.js:25]: I think you should combine this and the auth context. No need to have two providers

File: client/src/index.js [hunk 4]
```diff
@@ -4,9 +4,16 @@ import './index.css';
 import App from './App';
 import * as serviceWorker from './serviceWorker';
 
-ReactDOM.render(<App />, document.getElementById('root'));
+import AuthProvider from './context/AuthContext';
+import UserProvider from './context/UserContext';
+
+ReactDOM.render(
+  <AuthProvider>
+    <UserProvider>
```
Comment (aeciorc) [inline client/src/index.js:12]: Move the providers to App

File: client/src/components/balance/Topup.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    padding: '0.3rem',
+    background: '#dee2e6',
+    borderRadius: '5px',
+    margin: '0 1rem',
+    '&:hover': {
+      background: '#6E3ADB',
+      color: '#fff',
+    },
+  },
+  checkoutBtn: {
```
Comment (aeciorc) [inline client/src/components/balance/Topup.js:42]: Decrease the lateral paddings

File: client/src/components/balance/Balance.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+const Balance = () => {
+  const classes = useStyles();
+
+  const { user } = useContext(UserContext);
+
+  return (
+    <div>
+      <Paper elevation={0} className={classes.container}>
+        <Credit user={user} />
+        <Divider variant='middle' />
```
Comment (aeciorc) [inline client/src/components/balance/Balance.js:35]: Make the divider extend all the way, like in the mockup

File: client/src/components/balance/Topup.js [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+            <Remove className={classes.icon} />
+          </Button>
+          <Typography variant='h5' align='center'>
+            {topupCredit}
+          </Typography>
+          <Button
+            disableElevation
+            disableRipple
+            className={classes.iconBtn}
+            onClick={() => increase(topupCredit)}
```
Comment (aeciorc) [inline client/src/components/balance/Topup.js:105]: I like the purple hover effect, but the parent component is also getting a grey background during hover. Can you fix that?

**Review by aeciorc — CHANGES_REQUESTED** [review]

**Pull request closed without merging**